frontend/docsettings.lua

```diff
@@ -237,10 +282,19 @@ end
 function DocSettings:flush(data, no_custom_metadata)
     -- Depending on the settings, doc_settings are saved to the book folder or
     -- to koreader/docsettings folder. The latter is also a fallback for read-only book storage.
-    local serials = G_reader_settings:readSetting("document_metadata_folder", "doc") == "doc"
-        and { {self.doc_sidecar_dir, self.doc_sidecar_file},
+    local serials
+    local preferred_metdata_storage = G_reader_settings:readSetting("document_metadata_folder", "doc")
+    if preferred_metdata_storage == "doc" then
+        serials = { {self.doc_sidecar_dir, self.doc_sidecar_file},
               {self.dir_sidecar_dir, self.dir_sidecar_file}, }
-         or { {self.dir_sidecar_dir, self.dir_sidecar_file}, }
+    elseif preferred_metdata_storage == "dir" then
+        serials = { {self.dir_sidecar_dir, self.dir_sidecar_file}, }
+    elseif preferred_metdata_storage == "hash" then
+        serials = { table.pack(self:getSidecarHashDirAndFilepath(self.data.doc_path)) }
```

Comment:
You already have them from `open()` as `self.hash_sidecar_dir, self.hash_sidecar_file`